core/aws_ddk_core/stages/firehose_s3.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, List, Optional
+
+import aws_cdk.aws_kinesisfirehose_alpha as firehose
+import aws_cdk.aws_kinesisfirehose_destinations_alpha as destinations
+from aws_cdk import Duration, Size
+from aws_cdk.aws_events import EventPattern, IRuleTarget
+from aws_cdk.aws_kinesis import Stream
+from aws_cdk.aws_kms import IKey
+from aws_cdk.aws_logs import ILogGroup
+from aws_cdk.aws_s3 import IBucket
+from aws_ddk_core.pipelines.stage import DataStage
+from aws_ddk_core.resources import KinesisFactory, KinesisFirehoseFactory, S3Factory
+from constructs import Construct
+
+
+class FirehoseS3Stage(DataStage):
+    """
+    Class that represents a Firehose to S3 Ingestion DDK Stage.
+    """
+
+    def __init__(
+        self,
+        scope: Construct,
+        id: str,
+        environment_id: str,
+        delivery_stream_name: Optional[str] = None,
+        delivery_stream: Optional[firehose.IDeliveryStream] = None,
+        bucket_name: Optional[str] = None,
+        bucket: Optional[IBucket] = None,
+        buffering_interval: Optional[Duration] = None,
+        buffering_size: Optional[Size] = None,
+        compression: Optional[destinations.Compression] = None,
+        data_output_prefix: Optional[str] = None,
+        data_stream: Optional[Stream] = None,
+        enable_data_stream: Optional[bool] = False,
+        encryption_key: Optional[IKey] = None,
+        error_output_prefix: Optional[str] = None,
+        logging: Optional[bool] = True,
+        log_group: Optional[ILogGroup] = None,
+        alarm_threshold: Optional[int] = 900,
+        alarm_evaluation_periods: Optional[int] = 1,
+    ) -> None:
+        """
+        DDK Firehose to S3 stage.
```

Comment:
`DDK Kinesis Firehose Delivery stream to S3 stage, with an optional Kinesis Data Stream.`